File: src/java/ironfist/Trap.java
package ironfist;

/**
 * DOCUMENT ME!
 * 
 * @author pmgremo
 */
public class Trap {
  private boolean set;

  {
    set = true;
  }

  /**
   * DOCUMENT ME!
   */
  public void activate() {
    deactivate();
  }

  /**
   * DOCUMENT ME!
   *
   * @return DOCUMENT ME!
   */
  public boolean isSet() {
    return set;
  }

  /**
   * DOCUMENT ME!
   */
  public void deactivate() {
    set = false;
  }
}

File: src/main/java/ukalus/Trap.kt
package ukalus

/**
 * DOCUMENT ME!

 * @author pmgremo
 */
class Trap {

    /**
     * DOCUMENT ME!

     * @return DOCUMENT ME!
     */
    var isSet: Boolean = false
        private set

    init {
        isSet = true
    }

    /**
     * DOCUMENT ME!
     */
    fun activate() {
        deactivate()
    }

    /**
     * DOCUMENT ME!
     */
    fun deactivate() {
        isSet = false
    }
}